platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/preconsensus/DefaultInlinePcesWriterTest.java

```diff
@@ -0,0 +1,137 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Random;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class DefaultInlinePcesWriterTest {
+
+    @TempDir
+    private Path tempDir;
+
+    private final AncientMode ancientMode = GENERATION_THRESHOLD;
+    private PlatformContext platformContext;
+
+    @BeforeEach
+    void beforeEach() {
+        final Configuration configuration = new TestConfigBuilder()
+                .withValue(PcesConfig_.DATABASE_DIRECTORY, tempDir.toString())
+                .getOrCreateConfig();
+        platformContext = buildContext(configuration);
+    }
+
+    @NonNull
+    private PlatformContext buildContext(@NonNull final Configuration configuration) {
+        return TestPlatformContextBuilder.create()
+                .withConfiguration(configuration)
+                .withTime(new FakeTime(Duration.ofMillis(1)))
+                .build();
+    }
+
+    /**
+     * Build a transaction generator.
+     */
+    private static TransactionGenerator buildTransactionGenerator() {
+
+        final int transactionCount = 10;
+        final int averageTransactionSizeInKb = 10;
+        final int transactionSizeStandardDeviationInKb = 5;
+
+        return (final Random random) -> {
+            final TransactionWrapper[] transactions = new TransactionWrapper[transactionCount];
+            for (int index = 0; index < transactionCount; index++) {
+
+                final int transactionSize = (int) UNIT_KILOBYTES.convertTo(
+                        Math.max(
+                                1,
+                                averageTransactionSizeInKb
+                                        + random.nextDouble() * transactionSizeStandardDeviationInKb),
+                        UNIT_BYTES);
+                final byte[] bytes = new byte[transactionSize];
+                random.nextBytes(bytes);
+
+                transactions[index] = createAppPayloadWrapper(bytes);
+            }
+            return transactions;
+        };
+    }
+
+    /**
+     * Build an event generator.
+     */
+    static StandardGraphGenerator buildGraphGenerator(
+            @NonNull final PlatformContext platformContext, @NonNull final Random random) {
+        Objects.requireNonNull(platformContext);
+        final TransactionGenerator transactionGenerator = buildTransactionGenerator();
+
+        return new StandardGraphGenerator(
+                platformContext,
+                random.nextLong(),
+                new StandardEventSource().setTransactionGenerator(transactionGenerator),
+                new StandardEventSource().setTransactionGenerator(transactionGenerator),
+                new StandardEventSource().setTransactionGenerator(transactionGenerator),
+                new StandardEventSource().setTransactionGenerator(transactionGenerator));
+    }
```

Comment:
We should be able to replace all of this with the `TestingEventBuilder`

Reply:
No there is some issues with BIRTH_ROUND handling with the builder